### ISSUE TITLE
Wanted: a method of transferring stations between projects

### ISSUE THREAD

**Issue by j-sayers** [MotusDev/Motus-TO-DO#1312]
Placeholder for @cjardine-bsc

**Comment by cjardine-bsc** [comment]
What will happen to deployments within the station? Move as well? If there are deployments associated with the station that is far more complicated. 

Things impacted (for Catherine) 

- how to deal with sensor_id in sensor_deployments. This is tied to the project, so to move deployment we'd need to check to see if the same receiverID is registered to the new project and if so update sensor_id in the new project, if not then register the receiver to the destination project. 
-keep backup of deployments
-how to update det_batches_projects? Both removal of the old project and addition of the new project. 
-no change needed to det_taghits_deployId
-landowners are associated with station. Any permission issues.

**Comment by j-sayers** [comment]
Yeah I think any associated deployment should be moved as well (sorry!). It might be a newly created station and deployment but it also be one that has detections associated with it if it's been active for a while. 

If any associated receivers have only been affiliated with the Station in question, would it just be a matter of ensuring project_id is modified for both sensor and sensor_station (plus any downstream updates for derived tables)? If the receiver has been associated with other stations, though, then it would have to be "registered" anew... That seems too simple though!

Somewhat related is that we also need a method for transferring receivers (and their deployments) between stations. It could just be as "simple" as updating the sensor_id in sensor_stations if both receiver and station are in the same project. But when moving between projects there is the potential for mismatches.

There are actually a few places where project_id in sensors doesn't match the field in sensor_stations, which has created a few recently observed issues (in email, but soon to be repeated in a GitHub issue). 

Landowner permissions... It's potentially an issue but I think that nearly always stations will be moved between projects that likely share PIs or other members. Plus if the site needs to be visited in person they would already have to deal with introductions and and making sure the landowner knows there are new personnel involved. At least, that's their responsibility to do so. Personally I feel like the issue of permissions is sufficiently implied.

**Comment by cjardine-bsc** [comment]
-check if all sensor_ids within deployments for that station are used in any other stations within the project.
-for each sensor_id, if no change project_id in sensors table to new project. 
-if yes register receiver to new projects to get new sensor id (need to check if exists and pull existing sensor_id if so). 
-update all deployment for old project and station where sensor_id is changed.
-update project_id in station table

Landowners makes things complicated permission wise. to start strip out landowner ids from new station? 

Or, copy landowner ids to projects_landowners from sensor_stations_landowners
